File: src/lib/pjecalc/core/dominio/indices/ipca/indice-ipca.ts
/**
 * PJe-Calc v2.15.1 — IndiceIPCA
 * Porte 1:1 de: br.jus.trt8.pjecalc.negocio.dominio.indices.ipca.IndiceIPCA
 *
 * Ref Java: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/indices/ipca/IndiceIPCA.java
 */
import Decimal from 'decimal.js';
import { IndiceBase } from '../indice-base';
import { HelperDate } from '../../../base/comum/helper-date';
import type { Periodo } from '../../../base/comum/periodo';
import { calcularIndiceAcumulado } from '../../../comum/rotinasdecalculo/calculador-de-indices';
import { TABELA_IPCA } from './tabela-ipca';

export class IndiceIPCA extends IndiceBase {
  constructor(competencia: Date, taxa: Decimal, dataCriacao?: Date) {
    super(competencia, taxa, dataCriacao);
  }

  /** obterTabela (linha 78) */
  static obterTabela(periodo: Periodo): IndiceIPCA[] {
    const lista: IndiceIPCA[] = [];
    const inicio = HelperDate.getCurrentCompetence(periodo.getInicial());
    const fim = HelperDate.getCurrentCompetence(periodo.getFinal());
    for (const entry of TABELA_IPCA) {
      const comp = new Date(entry.ano, entry.mes - 1, 1);
      if (comp < inicio.getDate() || comp > fim.getDate()) continue;
      lista.push(new IndiceIPCA(comp, new Decimal(entry.taxa)));
    }
    lista.sort((a, b) => a.getCompetencia().getTime() - b.getCompetencia().getTime());
    return calcularIndiceAcumulado(lista) as IndiceIPCA[];
  }

  clonar(): IndiceIPCA {
    const c = new IndiceIPCA(this.competencia, this.taxa, this.dataCriacao);
    if (this.valorAcumulado !== null) c.setValorAcumulado(this.valorAcumulado);
    return c;
  }
}

File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/indices/ipca/IndiceIPCA.java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  javax.persistence.Column
 *  javax.persistence.Entity
 *  javax.persistence.GeneratedValue
 *  javax.persistence.GenerationType
 *  javax.persistence.Id
 *  javax.persistence.SequenceGenerator
 *  javax.persistence.Table
 *  org.jboss.seam.annotations.Name
 */
package br.jus.trt8.pjecalc.negocio.dominio.indices.ipca;

import br.jus.trt8.pjecalc.base.comum.Periodo;
import br.jus.trt8.pjecalc.negocio.dominio.indices.IndiceBase;
import br.jus.trt8.pjecalc.negocio.dominio.indices.ipca.RepositorioDeIndiceIPCA;
import java.math.BigDecimal;
import java.util.Date;
import java.util.List;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import org.jboss.seam.annotations.Name;

@Entity
@Table(name="TBIPCA")
@SequenceGenerator(name="SQIPCA", sequenceName="SQIPCA", allocationSize=1)
@Name(value="indiceIPCA")
public class IndiceIPCA
extends IndiceBase {
    private static final String NOME_INDICE = "IPCA";
    private static final long serialVersionUID = 6231191110416564771L;
    @Id
    @GeneratedValue(strategy=GenerationType.SEQUENCE, generator="SQIPCA")
    @Column(name="IIDINDICEIPCA", nullable=false)
    private Long id;

    public IndiceIPCA() {
        super(RepositorioDeIndiceIPCA.class);
    }

    public IndiceIPCA(Date competencia, BigDecimal taxa) {
        super(RepositorioDeIndiceIPCA.class, competencia, taxa);
    }

    @Override
    public Object obterChavePrimaria() {
        return this.getId();
    }

    public Long getId() {
        return this.id;
    }

    @Override
    public void salvar() {
        super.salvar();
    }

    public static List<IndiceIPCA> obterTodos() {
        return IndiceIPCA.getRepositorio(RepositorioDeIndiceIPCA.class).obterTodos();
    }

    public static List<IndiceIPCA> obterPorFiltro(IndiceIPCA filtro) {
        return IndiceIPCA.getRepositorio(RepositorioDeIndiceIPCA.class).obterPorFiltro(filtro);
    }

    public static void remover(IndiceIPCA entidade) {
        IndiceIPCA.remover(RepositorioDeIndiceIPCA.class, entidade, true);
    }

    public static List<IndiceIPCA> obterTabela(Periodo periodo) {
        return IndiceIPCA.getRepositorio(RepositorioDeIndiceIPCA.class).obterTabelaPor(periodo);
    }

    @Override
    public boolean existe() {
        return IndiceIPCA.getRepositorio(RepositorioDeIndiceIPCA.class).existe(this);
    }

    @Override
    public IndiceIPCA validar() {
        return (IndiceIPCA)this.validar(NOME_INDICE);
    }

    @Override
    public IndiceIPCA validarParaConsulta() {
        return (IndiceIPCA)this.validarParaConsulta(NOME_INDICE);
    }

    @Override
    public int hashCode() {
        return super.hashCode();
    }

    @Override
    public boolean equals(Object obj) {
        return super.equals(obj);
    }

    @Override
    public IndiceBase clonar() {
        IndiceIPCA indice = new IndiceIPCA();
        indice.id = this.getId();
        indice.setCompetencia(this.getCompetencia());
        indice.setTaxa(this.getTaxa());
        indice.setDataCriacao(this.getDataCriacao());
        return indice;
    }
}

